### ISSUE TITLE
Delete global cradle files (hie-cabal.yaml, hie-stack.yaml)

### ISSUE THREAD

**Issue by pepeiborra** [haskell/haskell-language-server#741]
I think this is great and I'm happy to merge it, but I have a couple of requests that can either be done in this PR of in follow-ups:

1. Inform the user that there is no cradle haskell/ghcide#788 
2. Write the implicit cradle to disk 
3. Give the user a choice if there are both stack and cabal project descriptors
4. Delete all the cradle files from ghcide and hls - no longer needed

_Originally posted by @pepeiborra in https://github.com/haskell/ghcide/pull/782#issuecomment-691618901_

**Comment by jneira** [comment]
maybe the files could be kept as an example of a complex project config?

**Comment by pepeiborra** [comment]
Can't `gen-hie` provide such an example?

**Comment by jneira** [comment]
Well, sort of, but the output is not exactly equal (it generates a path for each module in the exe component f.e.).
You have to install and execute the tool, of course, and you can check the file simply navigating the github repo.

**Comment by pepeiborra** [comment]
A file called `hie.foo.yaml` in our repository is a contract - we are telling other devs that this file can be used to load this repo in hls/ghcide. Even more than that, since we are the maintainers, we are telling them that this is *the way* to load this repo. And now we have made our life harder: we have to make sure to keep this file working and up to date through every change, add tests for it to make sure that it doesn't accidentally break when we upgrade `hie-bios` or change the layout of our repo, etc... 

All that for what? I personally don't use these files anymore since `gen-hie` became good enough. I don't use stack either, so I am not even able to regenerate the stack `hie.yaml` file. 

To summarise, by keeping them we are endorsing the wrong approach and making our lives harder in more than one way. So no, I don't think keeping them around is a good idea.

**Comment by ndmitchell** [comment]
Having dedicated examples is fine, even very desirable, but I think HLS should be using best practices, to encourage people to follow. I think we should now recommend not having a hie.yaml in most cases, since its easiest and works well. I've started to delete my hie.yaml files - they force a particular choice (Cabal vs Stack) which different devs make differently, and gen-hie seems to do better than my manual files in most cases.

**Comment by jneira** [comment]
Fair enough, i suppose implicit-hie is good enough to push eplicit hie.yaml to special corner cases

**Comment by jneira** [comment]
So we should remove them, however,  i would like to hear the opinion of the later "maintainers" of those files. We will delete the files if nobody disagree.
//cc @Ailrun @beberman @tittoassini @konn

**Comment by Ailrun** [comment]
So the dev flow will be use `gen-hie` to generate `hie.yaml` by dev self?

**Comment by jneira** [comment]
@Ailrun I think the idea is to use the implicit cradle and only generate it if you hit some corner case where it does not work
I assume that the implicit cradle works for hls.

**Comment by konn** [comment]
I think deleting *global* cradle makes sense, but deleting *all* cradles is not a good idea.
We need some `hie.yaml` to make `func-test` work as expected; for example, for functional test for Splice plugin, importing another module in the testdata required dedicated `hie.yaml` IIRC.

**Comment by Ailrun** [comment]
OK then, sounds good for me. (Though I haven't tried implicit cradle for HLS...)

**Comment by jneira** [comment]
> We need some hie.yaml to make func-test work as expected; for example, for functional test for Splice plugin, importing another module in the testdata required dedicated hie.yaml IIRC.

Yeah this is about the global ones, gonna change the description of the issue to make it clear

**Comment by pepeiborra** [comment]
One situation where it's unclear whether the implicit cradle works at all is when the hls wrapper is being used to detect the version of ghc before downloading binaries. I think this scenario requires further testing with the implicit cradle

**Comment by berberman** [comment]
I just tried loading HLS using implicit cradle, and it works smoothly. (though it seems that some components are recompiled repeatedly many times, which would be another issue not related to cradle)

**Comment by Ailrun** [comment]
OK, I have to admit I have agreed to delete global cradles too early. It looks like all plugin packages use Cabal cradle, as there is no `stack.yaml` next to their `.cabal` files.

**Comment by konn** [comment]
I have used hie.yaml.stack and global stack.yaml alone just worked; however, it turns out that deleting hie.yaml forces us to use Cabal as a backend, which forces rebuild to me...

**Comment by Ailrun** [comment]
I think using `gen-hie` is a way... As @pepeiborra, instead of managing global hie, let's recommend contributors to manage their own. Maybe we should put that somewhere in the contribution document.
